analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+
+public class Program
+{
+    public void Base(string[] myArray)
+    {
+        Method(new string[] { "s1", "s2" }); // Noncompliant {{Remove this array creation and simply pass the elements.}}
+//             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method(new string[] { "s1" }); // Noncompliant
+        Method("s1");           // Compliant
+        Method("s1", "s2");     // Compliant
+        Method(myArray);        // Compliant
+        Method(new string[12]); // Compliant
+
+        Method2(1, new string[] { "s1", "s2" }); // Noncompliant {{Remove this array creation and simply pass the elements.}}
+//                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method2(1, new string[] { "s1" }); // Noncompliant
+        Method2(1, "s1");           // Compliant
+        Method2(1, "s1", "s2");     // Compliant
+        Method2(1, myArray);        // Compliant
+        Method2(1, new string[12]); // Compliant
+
+        Method3(new string[] { "s1", "s2" }, "s1"); // Compliant
+        Method3(new string[] { "s1", "s2" }, new string[12]); // Compliant
+        Method3(new string[] { "s1", "s2" }, new string[] { "s1", "s2" }); // Noncompliant
+//                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method3(null, null);        // Compliant
+        Method4(new [] { "s1" });   // Compliant
+        Method4(new [] { "s1", "s2" }); // Compliant
+
+        Method3(args: new string[] { "s1", "s2" }, a: new string[12]); // Compliant (if you specifically require the arguments to be passed in this order there is no way of making this compliant, thus we shouldn't raise)
+        Method3(args: new string[12], a: new string[] { "s1", "s2" }); // Compliant
+
+        var s = new MyClass(1, new int[] { 2, 3 }); // Noncompliant
+//                             ^^^^^^^^^^^^^^^^^^
+        var s1 = new MyClass(1, 2, 3); // Compliant
+        s1 = new MyClass(args: new int[] { 2, 3 }, a: 1); // Compliant (if you specifically require the arguments to be passed in this order there is no way of making this compliant, thus we shouldn't raise)
+        var s2 = new MyOtherClass(args: new int[12], a: new int[] { 2, 3 }); // Compliant
+
+        var s3 = new IndexerClass();
+        var indexer1 = s3[new int[] { 1, 2 }]; // FN
```

Comment:
Can you try to add `SyntaxKind.ElementAccess` and see if it works? If not leave a comment here on what is missing and we will add an issue if needed.

Reply:
So `SyntaxKind.ElementAccessExpression` together with `ElementAccessExpressionSyntax `allows `GetLastArgumentIfArrayCreation `to work without any issue (`ArgumentListSyntax `parameter must be replaced with the common base class `BaseArgumentListSyntax `to accommodate also `BracketedArgumentListSyntax`).

The `IsParamParameter `method is the real deal. We cannot use the current structure because the symbol of `ElementAccessExpression` cannot be cast in `IMethodSymbol`, thus completing successfully the operation. 

Did I miss something? Any idea/workaround?

Reply:
> Did I miss something? Any idea/workaround?

No. It's fine. Baby steps. This is also about what I was expecting.

Please create an issue regarding  MethodParameterLookup. We need to add some overloads that except `IPropertySymbol` (I think that is the type you get in both the C# and VB cases). Nobody was paying attention to this loophole until now (which isn't as bad as it sounds because it's niche and we have False Negatives, and we treat FNs as acceptable in many cases).

Reply:
Did you create the issue? Please add the link to the issue here.

Reply:
https://github.com/SonarSource/sonar-dotnet/issues/6687